YMChat/src/main/java/com/yellowmessenger/ymchat/YMChat.java

```diff
@@ -93,10 +93,10 @@ public void closeBot() {
 
     public void emitEvent(YMBotEventResponse event) {
         if (event != null) {
-            if (botCloseEventListener != null && event.getCode() != null && event.getCode().equals("bot-closed")) {
+            if (botCloseEventListener != null && event.getCode() != null && isCloseBotEvent(event)) {
                 botCloseEventListener.onClosed();
             } else {
-                if (listener != null)
+                if (listener != null && !(isImageOpenEvent(event) || isImageCloseEvent(event)))
```

Comment:
@kaunteya  this is the check that will make sure, we don't pass "image-opened" and "image-closed" events to client.

Reply:
Hey @sukrit-ym can we have something generic to identify if the event is to be percolated to the host app or not? We can use a special parameter in the data part of the event, based on which we can filter such an event.

Reply:
That's a good point priyank. 
The entire info comes in "ymHandler" probably we can have another key at a top level saying "ymInternalEvent"

Reply:
done